2-mandatory/3-groceries.js

```diff
@@ -27,14 +27,34 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
+for (let day in weeklyMealPlan) {
+  for (let index = 0; index < day.length; index++) {
+    const element = day[index];
+    if (indexOf(element) === -1) {
```

Comment:
Also, I don't understand why you are said `indexOf ===-1` then push to the array. If the function indexOf returns -1 it means the element is not part of the array!